docs/protocol/core/v3.0.rst

```diff
@@ -0,0 +1,1164 @@
+raising an issue, please add the label
+"core-protocol-v3.0". Contributions and suggested edits can be made
+via GitHub via the `online editor
+<https://github.com/zarr-developers/zarr-specs/blob/core-protocol-v3.0-dev/docs/protocol/core/v3.0.rst>`_
+or by making a pull request against the "core-protocol-v3.0-dev"
+branch.
+
+This document was produced by the `Zarr core development team
+<https://github.com/orgs/zarr-developers/teams/core-devs>`_.
+
+
+Document conventions
+====================
+
+Conformance requirements are expressed with a combination of
+descriptive assertions and [RFC2119]_ terminology. The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in the normative
+parts of this document are to be interpreted as described in
+[RFC2119]_. However, for readability, these words do not appear in all
+uppercase letters in this specification.
+
+All of the text of this specification is normative except sections
+explicitly marked as non-normative, examples, and notes. Examples in
+this specification are introduced with the words "for example".
+
+
+Concepts and terminology
+========================
+
+This section introduces and defines some key terms and explains the
+conceptual model underpinning the Zarr protocol.
+
+.. _hierarchy:
+
+*Hierarchy*
+
+    A Zarr hierarchy is a tree structure, where each node in the tree
+    is either a group_ or an array_. Group nodes may have children but
+    array nodes may not. All nodes in a hierarchy have a name_ and a
+    path_.
+
+.. _group:
+.. _groups:
+
+*Group*
+
+    A group is a node in a hierarchy_ that may have child nodes.
+
+.. _array:
+.. _arrays:
+
+*Array*
+
+    An array is a node in a hierarchy_. An array is a data structure
+    with zero or more dimensions_ whose lengths define the shape_ of
+    the array. An array contains zero or more data elements_. All
+    elements_ in an array conform to the same `data type`_. An array
+    may not have child nodes.
+
+.. _name:
+.. _names:
+
+*Name*
+
+    Each node in a hierarchy_ has a name, which is a string of
+    characters with some additional constraints defined in the section
+    on `node names`_ below. Two sibling nodes cannot have the same
+    name. The root node does not have a name.
+
+.. _path:
+.. _paths:
+
+*Path*
+
+    Each node in a hierarchy_ has a path which uniquely identifies
+    that node and defines its location within the hierarchy_. The path
+    is formed by joining together the "/" character, followed by the
+    name_ of each ancestor node separated by the "/" character,
+    followed by the name_ of the node itself. For example, the path
+    "/foo/bar" identifies a node named "bar", whose parent is named
+    "foo", whose parent is the root of the hierarchy. The path "/"
+    identifies the root node.
+
+.. _dimension:
+.. _dimensions:
+
+*Dimension*
```

Comment:
I am slightly concerned that this may cause confusion with extensions of the zarr spec for dealing with named dimensions (I.e., netcdf). The term "dimension" in netcdf refers specifically to a named dimension (e.g., "time"), which may be shared between multiple arrays.

Reply:
Ah that's good to know. Not sure how best to resolve that, this use of "dimension" as in "multi-dimensional array" is also very common (e.g., [numpy](https://docs.scipy.org/doc/numpy-1.16.1/reference/arrays.ndarray.html)). Maybe a clarifying note here to say this is not a named dimension a la netcdf?